core/jazz_environment-event-handler/index.js

```diff
@@ -591,6 +624,214 @@ function triggerBuildJob(payload, serviceDetails, configData) {
   });
 }
 
+function createSafe(environmentPayload, service_id, configData, authToken) {
+  return new Promise((resolve, reject) => {
+    if (!configData.TVAULT && !configData.TVAULT.IS_ENABLED) {
+        logger.info('t-vault not enabled **', configData.TVAULT);
+      return resolve({
+        "error": "T-vault is not enabled",
+      });
+    }
+    var updatePayload = {};
+    const safeName = environmentPayload.service + '_' + environmentPayload.domain + '_' + environmentPayload.logical_id;
+    updatePayload.name = safeName;
+    updatePayload.owner = configData.SERVICE_USER;
+    updatePayload.description = "test safe";
+
+    var svcPayload = {
+      uri: configData.BASE_API_URL + '/' + environmentPayload.logical_id + configData.TVAULT.CREATE_SAFE_API_RESOURCE
+        + `?domain=${environmentPayload.domain}&service=${environmentPayload.service}&environment=${environmentPayload.logical_id}`,
+      method: "POST",
+      headers: {
+        "Authorization": authToken,
+        "Content-Type": "application/json",
+        "Jazz-Service-ID": service_id
+      },
+      json: updatePayload,
+      rejectUnauthorized: false
+    };
+    logger.info("safe svcpayload ****" + JSON.stringify(svcPayload));
+    request(svcPayload, function (error, response, body) {
+      if (response.statusCode && response.statusCode === 200) {
+        logger.info("response ***" + JSON.stringify(response));
+        logger.info("enviornmnet***" + JSON.stringify(environmentPayload));
+        const timestamp = new Date().toISOString();
+        const vaultLink = configData.TVAULT.HOSTNAME + "/#!/admin";
+        environmentPayload.safeDetails = { "safe": { "name": safeName, "ts": timestamp, "link": vaultLink } };
+        safeDetails = environmentPayload.safeDetails;
+        logger.info("environmentPayload createsafe" + JSON.stringify(environmentPayload));
+        return resolve(environmentPayload);
+      } else {
+        logger.error("Error creating safe: " + JSON.stringify(response));
+        return reject({
+          "error": "Error creating safe",
+          "details": response.body.message
+        });
+      }
+    });
+
+  });
+}
+
+function deleteSafe(environmentPayload, service_id, configData, authToken) {
```

Comment:
Please use a separate file for handling all the safe related activities

Reply:
As per discussion used wrapper function